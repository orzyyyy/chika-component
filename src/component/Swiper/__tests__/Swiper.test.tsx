import React from 'react';
import { mount } from 'enzyme';
import 'nino-cli/scripts/setup';
let Swiper;
switch (process.env.LIB_DIR) {
  case 'lib':
    Swiper = require('../../../../lib/component/Swiper').default;
    break;
  case 'dist':
    Swiper = require('../../../../dist/lib/Swiper').default;
    break;
  default:
    Swiper = require('..').default;
    break;
}

describe('Swiper', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('render correctly', () => {
    const wrapper = mount(
      <Swiper wrapperHeight={500} duration={0.7} sensibility={1}>
        test
      </Swiper>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('getChildHeight should work correctly', () => {
    const wrapper = mount(
      <Swiper wrapperHeight={500} duration={0.7} sensibility={1}>
        test
      </Swiper>,
    ).instance();
    wrapper.getChildHeight();
    expect(wrapper.bottomHeight).toBe(-456);
    expect(wrapper.scrollerHeight).toBe(0);
  });

  it('status loading', () => {
    const wrapper = mount(
      <Swiper wrapperHeight={500} duration={0.7} sensibility={1} loading>
        test
      </Swiper>,
    );
    const spy = jest.spyOn(wrapper.instance(), 'getChildHeight');
    wrapper.setState({ status: 'loading', type: 'refresh' });
    wrapper.setProps({ loading: false });
    wrapper.update();
    expect(wrapper.state('refreshText')).toBe('刷新成功');
    jest.runAllTimers();
    expect(wrapper.state('refreshText')).toBe('下拉刷新');
    expect(spy).toHaveBeenCalled();

    wrapper.setProps({ loading: true });
    wrapper.setState({ status: 'loading', type: 'load' });
    wrapper.update();

    wrapper.setProps({ loading: false });
    wrapper.update();
    expect(wrapper.state('loadText')).toBe('加载完成');
    jest.runAllTimers();
    expect(wrapper.state('loadText')).toBe('加载更多');
    expect(spy).toHaveBeenCalled();
  });

  it('touch event', () => {
    const onRefresh = jest.fn();
    const onLoad = jest.fn();
    const wrapper = mount(
      <Swiper
        wrapperHeight={500}
        duration={0.7}
        sensibility={1}
        loading
        onRefresh={onRefresh}
        onLoad={onLoad}
      >
        test
      </Swiper>,
    );
    wrapper
      .find('.wrapper')
      .simulate('touchstart', { touches: [{ pageY: 100 }] });
    expect(wrapper.instance().startY).toBe(100);

    wrapper
      .find('.wrapper')
      .simulate('touchmove', { touches: [{ pageY: 150 }] });
    expect(wrapper.state().distance).toBe(50);
    expect(wrapper.state().iconDeg).toBe(180);
    wrapper
      .find('.wrapper')
      .simulate('touchmove', { touches: [{ pageY: 130 }] });
    expect(wrapper.state().distance).toBe(30);
    expect(wrapper.state().iconDeg).toBe(122.72727272727272);
    wrapper
      .find('.wrapper')
      .simulate('touchmove', { touches: [{ pageY: 80 }] });
    expect(wrapper.state().distance).toBe(-20);
    expect(wrapper.state().iconDeg).toBe(180);

    wrapper.setState({ distance: 50 });
    wrapper.find('.wrapper').simulate('touchend');
    expect(onRefresh).toHaveBeenCalled();
    wrapper.setState({ distance: -10 });
    wrapper.find('.wrapper').simulate('touchend');
    expect(onLoad).toHaveBeenCalled();
  });
});
