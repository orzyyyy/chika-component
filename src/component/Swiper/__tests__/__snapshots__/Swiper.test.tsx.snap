// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper render correctly 1`] = `
<Swiper
  duration={0.7}
  sensibility={1}
  wrapperHeight={500}
>
  <div
    className="Swiper"
  >
    <div
      className="wrapper"
    >
      <div
        style={
          Object {
            "height": 500,
            "transform": "translate3d(0px, 0px, 0)",
          }
        }
      >
        test
      </div>
    </div>
  </div>
</Swiper>
`;
